Fix CIFuzz coverage by adding Jinja2 dependency.

oliverchang opened the pull request:
Also clean up the external action definitions:
- Set default values for the storage repo branches.
- Remove storage repo options from build_fuzzers.

jonathanmetzman commented on infra/cifuzz/external-actions/run_fuzzers/action.yml:
```diff
@@ -40,10 +40,12 @@ inputs:
     description: |
       The branch of the git repo to use for storing certain artifacts from
       fuzzing.
+    default: main
```
So now we are setting this by default instead of opting in. What are the consequences of this?

jonathanmetzman commented:
What does "Fix CIFuzz coverage by adding Jinja2 dependency" mean?
Fix checking test coverage? Or CIFuzz's ability to get cov info from projects it tests.

jonathanmetzman commented:
O I just noticed it means CIFuzz's ability to generate coverage reports.

oliverchang commented:
Updated this PR to fix the actual root cause, which is that we are copying our custom built Python3 from base-builder:

```
# Install newer Python from base-builder.
COPY --from=gcr.io/oss-fuzz-base/base-builder /usr/local/bin/python3 /usr/local/bin/python3
COPY --from=gcr.io/oss-fuzz-base/base-builder /usr/local/lib/libpython3* /usr/local/lib/
COPY --from=gcr.io/oss-fuzz-base/base-builder /usr/local/lib/python3.8 /usr/local/lib/python3.8
RUN ldconfig
```

This is a problem because:
- The base-builder one is actually *older* (3.8.3 vs 3.8.10)
- We do actually `pip install` the requirements for code coverage, but that `pip` comes from Ubuntu and installs things to  the "dist-packages" which isn't recognized by our custom built Python.

I removed the Python copying part in this PR, but we might want to remove that from the base-builder as well.

jonathanmetzman commented:
> Updated this PR to fix the actual root cause, which is that we are copying our custom built Python3 from base-builder:
> 
> ```
> # Install newer Python from base-builder.
> COPY --from=gcr.io/oss-fuzz-base/base-builder /usr/local/bin/python3 /usr/local/bin/python3
> COPY --from=gcr.io/oss-fuzz-base/base-builder /usr/local/lib/libpython3* /usr/local/lib/
> COPY --from=gcr.io/oss-fuzz-base/base-builder /usr/local/lib/python3.8 /usr/local/lib/python3.8
> RUN ldconfig
> ```
> 
> This is a problem because:
> 
> * The base-builder one is actually _older_ (3.8.3 vs 3.8.10)
> * We do actually `pip install` the requirements for code coverage (which includes Jinja2), but that `pip` comes from Ubuntu and installs things to  the "dist-packages" which isn't recognized by our custom built Python.
> 
> I removed the Python copying part in this PR, but we might want to remove that from the base-builder as well in another PR.

Ah good. I wanted to get rid of this hack, but wasn't sure if it would cause issues. Thanks!

jonathanmetzman reviewed (approved):
LGTM
